src/MessagePack.GeneratorCore/Generator/FormatterTemplate.tt

```diff
@@ -21,101 +21,110 @@
         {
             if (reader.TryReadNil())
             {
-<# if( objInfo.IsClass) { #>
+<# if (objInfo.IsClass) {#>
                 return null;
-<# } else { #>
-                throw new InvalidOperationException("typecode is null, struct not supported");
-<# } #>
+<# } else {#>
+                throw new global::System.InvalidOperationException("typecode is null, struct not supported");
+<# }#>
             }
 
+<# if (isFormatterResolverNecessary) {#>
+            var formatterResolver = options.Resolver;
+<# }#>
             options.Security.DepthStep(ref reader);
-<# if (isFormatterResolverNecessary) { #>
-            IFormatterResolver formatterResolver = options.Resolver;
-<# } #>
             var length = reader.ReadArrayHeader();
-<# foreach(var x in objInfo.Members) { #>
-            var __<#= x.Name #>__ = default(<#= x.Type #>);
-<# } #>
+<# foreach (var member in objInfo.Members) {#>
+            var __<#= member.Name #>__ = default(<#= member.Type #>);
+<# }#>
 
-            for (int i = 0; i < length; i++)
+            for (var i = 0; i < length; i++)
             {
                 switch (i)
                 {
-<# foreach(var x in objInfo.Members) { #>
-                    case <#= x.IntKey #>:
-                        __<#= x.Name #>__ = <#= x.GetDeserializeMethodString() #>;
+<# for (var memberIndex = 0; memberIndex <= objInfo.MaxKey; memberIndex++) {
+  var member = objInfo.GetMember(memberIndex);
+  if (member == null) { continue; }#>
+                    case <#= member.IntKey #>:
+                        __<#= member.Name #>__ = <#= member.GetDeserializeMethodString() #>;
                         break;
-<# } #>
+<# }#>
                     default:
                         reader.Skip();
                         break;
                 }
             }
 
             var ____result = new <#= objInfo.GetConstructorString()  #>;
-<# foreach(var x in objInfo.Members.Where(x => x.IsWritable)) { #>
-            ____result.<#= x.Name #> = __<#= x.Name #>__;
-<# } #>
-<#if(objInfo.HasIMessagePackSerializationCallbackReceiver && objInfo.NeedsCastOnAfter) { #>
-            ((IMessagePackSerializationCallbackReceiver)____result).OnAfterDeserialize();
-<# } else if(objInfo.HasIMessagePackSerializationCallbackReceiver) { #>
+<# for (var memberIndex = 0; memberIndex <= objInfo.MaxKey; memberIndex++) {
+  var member = objInfo.GetMember(memberIndex);
+  if (member == null || !member.IsWritable) { continue; }#>
+            ____result.<#= member.Name #> = __<#= member.Name #>__;
+<# }
+ if (objInfo.HasIMessagePackSerializationCallbackReceiver) {
+  if (objInfo.NeedsCastOnAfter) {#>
+            ((global::MessagePack.IMessagePackSerializationCallbackReceiver)____result).OnAfterDeserialize();
+<#  } else {#>
             ____result.OnAfterDeserialize();
-<# } #>
+<#  }
+ }#>
             reader.Depth--;
             return ____result;
         }
     }
-<# } #>
+<#}#>
```

Comment:
I prefer the whitespace. Anything right up against `<#` or `#>` is harder to read. Please revert all such changes.

Reply:
I agree with you.